src/main/java/com/exadel/discountwebapp/validation/VendorValidator.java

```diff
@@ -15,6 +17,7 @@ public void validate(VendorRequestVO request) {
         checkDuplicateEmail(request);
     }
 
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
     public void checkDuplicateEmail(VendorRequestVO request) {
         vendorRepository.findByEmail(request.getEmail())
```

Comment:
you can use vendorRepository.existsById instead

Reply:
added existsByEmail() method here and in repo